safe/load.go

```diff
@@ -3,117 +3,89 @@ package safe
 import (
 	"encoding/json"
 	"fmt"
-	"reflect"
 
 	"github.com/bndw/pick/backends"
 	"github.com/bndw/pick/config"
 	"github.com/bndw/pick/crypto"
 	"github.com/bndw/pick/errors"
-	"github.com/bndw/pick/utils"
 )
 
+// Load loads, decrypts, and returns an instantiated Safe.
+// In the case that the Safe configuration as changed, the Safe will be upgraded
+// to the latest configuration and saved.
 func Load(password []byte, backend backends.Client, encryptionClient crypto.Client, config *config.Config) (*Safe, error) {
-	s := Safe{
-		backend:  backend,
-		crypto:   encryptionClient,
-		password: password,
-		Config:   config,
-	}
-
-	data, err := s.backend.Load()
+	ciphertext, err := backend.Load()
 	if err != nil {
-		if err == errors.ErrSafeNotFound {
-			s.Accounts = make(map[string]Account)
-			s.Notes = newNotesManager(&s)
-			return &s, nil
+		switch err {
+		case errors.ErrSafeNotFound:
+			return NewSafe(password, backend, encryptionClient, config, nil, nil)
+		default:
+			return nil, err
 		}
-		return nil, err
-	}
-
-	safeDTO := safeDTO{}
-	if err := json.Unmarshal(data, &safeDTO); err == nil { // nolint: vetshadow
-		// Unmarshalling did succeed -> Safe uses new format
-		// Do nothing
-	} else {
-		safeDTO.Ciphertext = data
-		// If we don't have a config, use OpenPGP for backwards-compatibility
-		defaultOpenPGPConfig := crypto.NewDefaultConfigWithType(crypto.ConfigTypeOpenPGP)
-		safeDTO.Config = &defaultOpenPGPConfig
-	}
-
-	// We first try to decrypt a safe using its own configuration
-	s.crypto, err = crypto.New(safeDTO.Config)
-	if err != nil {
-		return nil, err
 	}
 
-	upgradeCrypto := encryptionClient
-	plaintext, err := s.crypto.Decrypt(safeDTO.Ciphertext, password)
+	var configurationHasChanged bool
```

Comment:
Haha, I'd initially named it `updateRequired`. 

The reason I changed it to `configurationHasChanged` was because I felt it was more flexible; as a result of changing configuration we may need to perform other actions, for example, in addition to an upgrade. 

If you feel it's too vague or confusing, I'm willing to change it back to `upgradeRequired`.

Reply:
> If you feel it's too vague or confusing

The reason I want it changed is that it's too long 😹 What do you think of `configChanged`?